src/Build/BuildCheck/Infrastructure/EditorConfig/EditorConfigParser.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Drawing.Design;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Build.Shared;
+using static Microsoft.Build.BuildCheck.Infrastructure.EditorConfig.EditorConfigGlobsMatcher;
+
+namespace Microsoft.Build.BuildCheck.Infrastructure.EditorConfig
+{
+    internal sealed class EditorConfigParser
+    {
+        private const string EditorconfigFile = ".editorconfig";
+
+        /// <summary>
+        /// Cache layer of the parsed editor configs the key is the path to the .editorconfig file.
+        /// </summary>
+        private readonly ConcurrentDictionary<string, EditorConfigFile> _editorConfigFileCache = new ConcurrentDictionary<string, EditorConfigFile>(StringComparer.InvariantCultureIgnoreCase);
+
+        internal Dictionary<string, string> Parse(string filePath)
+        {
+            var editorConfigs = DiscoverEditorConfigFiles(filePath);
+            return MergeEditorConfigFiles(editorConfigs, filePath);
+        }
+
+        /// <summary>
+        /// Fetches the list of EditorconfigFile ordered from the nearest to the filePath.
+        /// </summary>
+        /// <param name="filePath"></param>
+        internal List<EditorConfigFile> DiscoverEditorConfigFiles(string filePath)
+        {
+            var editorConfigDataFromFilesList = new List<EditorConfigFile>();
+
+            var directoryOfTheProject = Path.GetDirectoryName(filePath);
+            var editorConfigFilePath = FileUtilities.GetPathOfFileAbove(EditorconfigFile, directoryOfTheProject);
```

Comment:
Why don't we start with the directory where the project is located?

Reply:
We are starting in the project's directory. 
The way GetPathOfFileAbove works:  Accepts file and starting Directory, if found returns the path, if not empty string. (Added the comment line in the code)
// The method will look for the file in parent directory if not found in current until found or the directory is root.

Reply:
Oh I see. Thx for claryfying comment then!